pf/tests/provider_create_test.go

```diff
@@ -174,3 +174,61 @@ func TestCreateWithFirstClassSecrets(t *testing.T) {
 	}`
 	testutils.Replay(t, server, testCase)
 }
+
+func TestCreateWithSchemaBasedSecrets(t *testing.T) {
+	// Ensure that resources that mark output properties as secret in the schema return them as secrets.
+	// RandomPassword is a good example. Surprisingly this test requires a Configure call first, otherwise the
+	// plubming is confused about secrets bits and retursn the wrong result. The test represents production use.
+	server := newProviderServer(t, testprovider.RandomProvider())
+	testCase := `
+	[
+	  {
+	    "method": "/pulumirpc.ResourceProvider/Configure",
+	    "request": {
+	      "args": {},
+	      "acceptSecrets": true,
+	      "acceptResources": true
+	    },
+	    "response": {
+	      "acceptSecrets": true,
+	      "supportsPreview": true,
+	      "acceptResources": true
+	    }
```

Comment:
The response here actually tests our `newProviderServer`. It doesn't get to code in this repo. The rest is good.

Reply:
Yeah, I can mark it "*". Looks like /pulumirpc.ResourceProvider/Configure seeds the plumbing to get ready to send secrets back. We don't care about the response.